easybuild/easyblocks/g/gcc.py

```diff
@@ -891,20 +898,81 @@ def post_install_step(self, *args, **kwargs):
                     "This directory was renamed by EasyBuild to avoid that the header files in it are picked up,",
                     "since they may cause problems when the OS is upgraded to a new (minor) version.",
                     '',
-                    "These files were copied to %s first: %s" % (include_path, ', '.join(retained_header_files)),
+                    "These files were copied to %s first: %s" % (include_fixed_path, ', '.join(retained_header_files)),
                     '',
                     "See https://github.com/easybuilders/easybuild-easyconfigs/issues/10666 for more information.",
                     '',
                 ])
                 write_file(readme, readme_txt)
 
-                include_fixed_renamed = include_fixed_path + '.renamed-by-easybuild'
-                move_file(include_fixed_path, include_fixed_renamed)
-                self.log.info("%s renamed to %s to avoid using the header files in it",
-                              include_fixed_path, include_fixed_renamed)
-            else:
-                raise EasyBuildError("Exactly one 'include-fixed' directory expected, found %d: %s",
-                                     len(include_fixed_paths), include_fixed_paths)
+            # If the mkheaders utility exists we create a wrapper script to (re)generate the fixed headers manually,
+            # e.g. after OS upgrades.
+            # Get the '*-linux-gnu' part
+            target_machine = os.path.basename(os.path.dirname(os.path.dirname(include_fixed_path)))
+            mkheaders_path = os.path.join(self.installdir, 'libexec', 'gcc', target_machine, self.version,
+                                          'install-tools', 'mkheaders')
+            if os.path.isfile(mkheaders_path):
+                recreate_include_fixed_script = os.path.join(self.installdir, 'easybuild', 'recreate_includes.sh')
+                self.log.info("Creating script to regenerate %s on OS upgrades at %s",
+                              include_fixed_path, recreate_include_fixed_script)
+                relative_mkheaders = os.path.relpath(mkheaders_path, self.installdir)
+                relative_include_fixed = os.path.relpath(include_fixed_path, self.installdir)
+                makedirs(os.path.dirname(recreate_include_fixed_script))
+                write_file(recreate_include_fixed_script, textwrap.dedent("""\
+                    #!/bin/bash
```

Comment:
I prefer if we make this whole thing a constant at the top of the easyblock, so here we can simply do:

```python
write_file(recreate_include_fixed_script, RECREATE_INCLUDE_FIXED_SCRIPT_BODY)